### PR TITLE
Show live tournament in China depending on time and Mandate setting

### PR DESCRIPTION
**Pull request by nwinter**
This goes alongside the server-side tournament-level-access PR, although neither should break if the other hasn't been merged.

This lets us use the Mandate to control when a live tournament would start (show access to the tournament level on a student dashboard) and stop (kick the students out of the level if they try to play when it's not active), to within two minutes, independent of client time settings.

For students in the target tournament classroom, it also shows details about the tournament if within a few day window of the tournament: either it's inactive (hasn't started or has already stopped), and what the scoring is like. This depends on client-side time, so it's not very precise.

It's a bit janky, hard-coding the tournament arena name and the target authorized classroom id in a few places, but I don't have time to increase the level of abstractions–a lot of this is reused from a previous live tournament setup we did for Israel.

Before/after tournament, within the specified date range:
Screenshot 2019-06-19 12 06 29

During tournament:
Screenshot 2019-06-19 11 58 43

Rules hover:
Screenshot 2019-06-19 12 08 10

To do: write rules, get Chinese translations for rules and a few new UI strings, get a legit Tarena tournament logo image.

### CODE REVIEW

File: app/templates/courses/courses-view.jade [hunk 1]
```diff
@@ -37,6 +37,44 @@ block content
...
+                a#tournament-rules(data-i18n="ladder.tournament_rules")
+                #tournament-rules-content.hide.rtl-allowed
+                  if /^zh/.test(me.get('preferredLanguage', true))
+                    p Welcome to the Tarena CodeCombat live tournament! Compete for the top 200 spots to qualify for the next round.
+
+                    p Scoring is as follows: your code will play constantly in the background against other players to determine how strong it is. The number of points you earn for a match are determined by a Bayesian calculation predicting how likely you were to win that match. So if you beat a player ranked more highly than you, you gain more points. Beat a player lower than you, gain fewer points. Lose to a weaker player, lose more points. Lose to a stronger player, lose fewer points. Your code will play on both the red and blue teams, but scoring will be based on the red team.
+
+                    p You can test your code against any player and improve your strategy until you win, but remember: no points are scored until you submit your latest code and it starts to automatically plays matches against other players.
+                  else
```
Comment (shubhi1092) [inline app/templates/courses/courses-view.jade]: Is there any difference between the text in the above condition (lines 48 - 52), vs below at line 54-58? They look the same - do we need if-else here?

File: app/views/courses/CoursesView.coffee [hunk 2]
```diff
@@ -119,6 +120,25 @@ module.exports = class CoursesView extends RootView
         @render()
       @supermodel.trackRequest(levels.fetchForClassroom(classroomID, { data: { project: "original,primerLanguage,slug,i18n.#{me.get('preferredLanguage', true)}" }}))
 
+    if features.china and @classrooms.find {id: '5d0082964ebb960059fc40b2'}
+      if new Date() >= new Date(2019, 5, 19, 12) && new Date() <= new Date(2019, 5, 25, 0)
+        if window.serverConfig?.currentTournament or true
```
Comment (shubhi1092) [inline app/views/courses/CoursesView.coffee]: This seems to be always true, was this (`or true`) left from debugging?

Comment (nwinter) [reply app/views/courses/CoursesView.coffee]: Missed that, thanks! Removed.

File: app/templates/courses/courses-view.jade (hunk 1 above)
Comment (nwinter) [reply app/templates/courses/courses-view.jade]: Right, this is just a placeholder for now; when I get Chinese translations from the China team, I'll put them in here.

Comment (shubhi1092) [reply app/templates/courses/courses-view.jade]: Should this be moved to en.coffee then?

**Pull request merged**